src/aws/auth.rs

```diff
@@ -28,7 +30,10 @@ pub enum AwsAuthentication {
 }
 
 impl AwsAuthentication {
-    pub async fn credentials_provider(&self) -> crate::Result<SharedCredentialsProvider> {
+    pub async fn credentials_provider(
+        &self,
+        region: Option<Region>,
```

Comment:
This might end up needing to be non-`Option`al given the other AWS PR in flight.

Reply:
Yeah, depending on which of these is merged first, I'll update the other.